chore: update actions

ryjones opened the pull request:
actions/upload-artifact and actions/download-artifact have been deprecated and need updated.

https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

survived commented:
Thanks for the PR! We have a policy that prevents us from merging unsigned commits @ryjones. Can you sign them? Or alternatively I can re-do PR with my commits

ryjones commented:
@survived I can but it would break your commit signature, so please go ahead

ryjones committed:
chore: update actions

actions/upload-artifact and actions/download-artifact have
been deprecated and need updated.

https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

Signed-off-by: Ry Jones <ry@linux.com>

ryjones commented:
@survived I added your changes to my commit

survived commented:
It's strange that my review was not dismissed after you pushed a new commit despite the protection settings of the branch. Probably a github bug?

survived commented:
The diff is the same as approved one, so maybe github didn't treat the new commit as something that should dismiss the review approval

ryjones commented:
@survived I updated [the ruleset](https://github.com/LFDT
<img width="715" alt="Screenshot 2025-01-10 at 08 35 21" src="https://github.com/user-attachments/assets/bb1302be-3d46-454c-bd94-02aa3c62716f" />
-Lockness/cggmp21/settings/rules/2940679)

survived commented:
I don't think it will work actually, all it do (from my experience) is that if I do the most recent push, I can't approve PR

survived commented:
in this case, you did the most recent push, I did the review, this policy does not apply (I also think it has confusing name)
